src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -59,7 +59,7 @@ impl<F: FieldExt> CondSwapInstructions<F> for CondSwapChip<F> {
     fn swap(
         &self,
         mut layouter: impl Layouter<F>,
-        pair: (Self::Var, Self::Var),
+        pair: (Self::Var, Option<F>),
```

Comment:
Document in CondSwapInstructions that we require the second element of the pair to be just a witnessed value, not something that is copied from elsewhere.